Src/EmbeddedScripts.JS.ClearScriptV8/ClearScriptV8Runner.cs

```diff
@@ -1,24 +1,39 @@
 ﻿using System.Threading.Tasks;
 using EmbeddedScripts.Shared;
+using Microsoft.ClearScript;
 using Microsoft.ClearScript.V8;
 
 namespace EmbeddedScripts.JS.ClearScriptV8
 {
     public class ClearScriptV8Runner : ICodeRunner
     {
-        private Container container = new();
+        private Container _container = new();
 
-        public async Task RunAsync(string code)
+        public Task RunAsync(string code)
         {
             using var engine = new V8ScriptEngine();
-            engine.AddHostObjectsFromContainer(container);
+            engine.AddHostObjectsFromContainer(_container);
 
-            await Task.Run(() => engine.Execute(code));
+            try
+            {
+                engine.Execute(code);
+            }
+            catch (ScriptEngineException e)
+            {
+                if (e.Message.StartsWith("SyntaxError"))
+                    throw new ScriptSyntaxErrorException("Syntax error", e);
+
+                var doubleInnerException = e.InnerException?.InnerException?.InnerException; 
```

Comment:
a more common way is to get the deepest inner exception instead: `while (e.InnerException != null) e = e.InnerException;`

Reply:
Here I need exception exactly at this depth, because if some exposed function throws an exception, user code can access all inner exceptions inside this one.